tests/components/jewish_calendar/test_binary_sensor.py

```diff
@@ -0,0 +1,98 @@
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.const import STATE_ON, STATE_OFF
+import homeassistant.util.dt as dt_util
+from homeassistant.setup import async_setup_component
+from homeassistant.components import jewish_calendar
+
+from tests.common import async_fire_time_changed
+
+from . import make_nyc_test_params, make_jerusalem_test_params
+
+melacha_params = [
+    make_nyc_test_params(dt(2018, 9, 1, 16, 0), STATE_ON),
+    make_nyc_test_params(dt(2018, 9, 1, 20, 21), STATE_OFF),
+    make_nyc_test_params(dt(2018, 9, 7, 13, 1), STATE_OFF),
+    make_nyc_test_params(dt(2018, 9, 8, 21, 25), STATE_OFF),
+    make_nyc_test_params(dt(2018, 9, 9, 21, 25), STATE_ON),
+    make_nyc_test_params(dt(2018, 9, 10, 21, 25), STATE_ON),
+    make_nyc_test_params(dt(2018, 9, 28, 21, 25), STATE_ON),
+    make_nyc_test_params(dt(2018, 9, 29, 21, 25), STATE_OFF),
+    make_nyc_test_params(dt(2018, 9, 30, 21, 25), STATE_ON),
+    make_nyc_test_params(dt(2018, 10, 1, 21, 25), STATE_ON),
+    make_jerusalem_test_params(dt(2018, 9, 29, 21, 25), STATE_OFF),
+    make_jerusalem_test_params(dt(2018, 9, 30, 21, 25), STATE_ON),
+    make_jerusalem_test_params(dt(2018, 10, 1, 21, 25), STATE_OFF),
+]
+
+melacha_test_ids = [
+    "currently_first_shabbat",
+    "after_first_shabbat",
+    "friday_upcoming_shabbat",
+    "upcoming_rosh_hashana",
+    "currently_rosh_hashana",
+    "second_day_rosh_hashana",
+    "currently_shabbat_chol_hamoed",
+    "upcoming_two_day_yomtov_in_diaspora",
+    "currently_first_day_of_two_day_yomtov_in_diaspora",
+    "currently_second_day_of_two_day_yomtov_in_diaspora",
+    "upcoming_one_day_yom_tov_in_israel",
+    "currently_one_day_yom_tov_in_israel",
+    "after_one_day_yom_tov_in_israel",
+]
+
+
+@pytest.mark.parametrize(
+    [
+        "now",
+        "candle_lighting",
+        "havdalah",
+        "diaspora",
+        "tzname",
+        "latitude",
+        "longitude",
+        "result",
+    ],
+    melacha_params,
+    ids=melacha_test_ids,
+)
+async def test_issur_melacha_sensor(
+    hass, now, candle_lighting, havdalah, diaspora, tzname, latitude, longitude, result
+):
+    """Test Issur Melacha sensor output."""
+    time_zone = dt_util.get_time_zone(tzname)
+    test_time = time_zone.localize(now)
+
+    hass.config.time_zone = time_zone
+    hass.config.latitude = latitude
+    hass.config.longitude = longitude
+
+    assert await async_setup_component(
+        hass,
+        jewish_calendar.DOMAIN,
+        {
+            "jewish_calendar": {
+                "name": "test",
+                "language": "english",
+                "diaspora": diaspora,
+                "candle_lighting_minutes_before_sunset": candle_lighting,
+                "havdalah_minutes_after_sunset": havdalah,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    with patch("homeassistant.util.dt.now", return_value=test_time):
+        future = dt_util.utcnow() + timedelta(seconds=30)
```

Comment:
We also need to patch util.dt.utcnow since that's used by the event helper used in the entity platform helper.

Reply:
But I don't really care for `utcnow()` I need to patch `now()` because that's what the entity uses. The only reason I see to patch both, is if at a future time the entity helper will be using `now()` instead and then we'll have a mismatch.

Reply:
We need to control the time that is used to check when an update should occur. So patch utcnow before setting up the platform and adding the entity.

Reply:
Since I set the future to be the default value of SCAN_INTERVAL in the future shouldn't that be enough?
I fire an event that time is now `SCAN_INTERVAL` seconds in the future, so the listener should detect that.

Reply:
We should take for habit to patch time in tests where we need to control time to avoid outside interference, eg slow test hardware. Otherwise there will be a time difference between when we set up the platform and when we capture time before firing the event. This might not matter for all tests, but it's better to have the correct approach to avoid flaky tests.

Reply:
Ok, makes sense now. Thanks for taking the time to explain it to me.
So should I reinstate the alter_time function I had before? Or is there a better way to implement that?

Reply:
I think that was a good way.

Reply:
Done